lib/src/file.dart

```diff
@@ -49,15 +49,28 @@ class SourceFile {
   /// previous result.
   int _cachedLine;
 
-  /// Creates a new source file from [text].
+  /// This constructor is deprecated.
   ///
-  /// [url] may be either a [String], a [Uri], or `null`.
+  /// Use [new SourceFile.fromString] instead.
+  @Deprecated("Will be removed in 2.0.0")
```

Comment:
[nit] would it be useful to put a reference to SourceFile.fromString in the deprecated message?